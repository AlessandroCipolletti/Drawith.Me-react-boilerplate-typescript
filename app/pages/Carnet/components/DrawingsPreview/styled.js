import Config from 'common/Config'
import styled, { css } from 'styled-components'

import draftIcon from 'images/icons/draft.png'
import selectIcon from 'images/icons/select.png'
import paper from 'images/papers/white.png'

const DrawingCssPhone = css`
  @media (orientation: landscape) {
    width: 30vw;
    height: 35vh;
  }
  @media (orientation: portrait) {
    width: 44vw;
    height: 16vh;
  }
  margin: ${({ theme }) => theme.spacing.size2} 0px;
`

const DrawingCssTablet = css`
  @media (orientation: landscape) {
    width: 20vw;
    height: 18.5vh;
  }
  @media (orientation: portrait) {
    width: 29vw;
    height: 15vh;
  }
  margin: ${({ theme }) => theme.spacing.size4} 0px;
`

const DrawingCssDesktop = css`
  width: 18vw;
  height: 21vh;
  margin: ${({ theme }) => theme.spacing.size4} 0px;
`

export const DrawingCss = css`
  ${Config.isDesktop ? DrawingCssDesktop : ''}
  ${Config.isTablet ? DrawingCssTablet : ''}
  ${Config.isPhone ? DrawingCssPhone : ''}
  border: solid 1px rgb(220, 220, 220);
  border-radius: ${({ theme }) => theme.spacing.size2};
  box-shadow: inset 0px 0px ${({ theme }) => theme.spacing.size1} 1px rgba(220, 220, 220, 0.5);
  cursor: pointer;
`

export const Wrapper = styled.div`
  ${DrawingCss}
  background: url(${props => props.drawing}) no-repeat center, url(${paper});
  background-size: contain, auto;
`

export const DraftLabel = styled.div`
  position: relative;
  display: inline-block;
  top: 0px;
  left: 0px;
  width: ${({ theme }) => theme.spacing.size7};
  height: ${({ theme }) => theme.spacing.size7};
  border-radius: ${({ theme }) => theme.spacing.size2} 0px;
  border-right: solid 1px ${({ theme }) => theme.palette.gray[6]};
  border-bottom: solid 1px ${({ theme }) => theme.palette.gray[6]};
  background-color: ${({ theme }) => theme.palette.gray[8]};
  background-size: 70%;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url(${draftIcon});
`

export const SelectLabel = styled.div`
  float: right;
  display: inline-block;
  top: 0px;
  right: 0px;
  margin: ${({ theme }) => theme.spacing.size2};
  height: ${({ theme }) => theme.spacing.size5};
  width: ${({ theme }) => theme.spacing.size5};
  border: solid 1px white;
  background-color: white;
  border-radius: 50%;
  background-size: 60%;
  background-position: center;
  background-repeat: no-repeat;
  box-shadow: 0px 0px 2px 0px rgba(0, 0, 0, 0.75);
  background-color: ${({ selected }) => selected ? 'rgb(25, 110, 225)' : 'none'};
  background-image: ${({ selected }) => selected ? `url(${selectIcon})` : 'none'};
`
