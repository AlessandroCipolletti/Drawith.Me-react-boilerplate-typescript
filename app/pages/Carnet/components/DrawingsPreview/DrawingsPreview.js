import React from 'react'
import PropTypes from 'prop-types'

import { Wrapper, DraftLabel, SelectLabel } from './styled'

const DrawingsPreview = ({
  id, drawing, selectMode = false, selected = false, draft = false, onClick = () => {},
}) => {
  const handleClick = React.useCallback(() => {
    onClick(id)
  }, [id, onClick])

  return (
    <Wrapper
      data-id={id}
      drawing={drawing}
      onClick={handleClick}
    >
      {draft && <DraftLabel />}
      {selectMode && <SelectLabel selected={selected} />}
    </Wrapper>
  )
}

DrawingsPreview.propTypes = {
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  drawing: PropTypes.string.isRequired,
  selectMode: PropTypes.bool,
  selected: PropTypes.bool,
  draft: PropTypes.bool,
  onClick: PropTypes.func,
}

export default DrawingsPreview
