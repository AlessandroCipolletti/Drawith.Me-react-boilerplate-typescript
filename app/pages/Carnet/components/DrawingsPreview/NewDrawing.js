import styled from 'styled-components'


import newImg from 'images/UI/Carnet/new.png'
import paper from 'images/papers/white.png'
import { DrawingCss } from './styled'

const NewDrawing = styled.div`
  ${DrawingCss}
  background: url(${newImg}) no-repeat center, url(${paper});
  background-size: 45%, auto;
  opacity: ${props => props.selectMode ? 0.3 : 1};
  cursor: ${props => props.selectMode ? 'auto' : 'pointer'};
`

export default NewDrawing
