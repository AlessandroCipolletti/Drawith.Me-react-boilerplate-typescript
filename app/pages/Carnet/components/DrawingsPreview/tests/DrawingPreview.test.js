import React from 'react'
import { render, fireEvent } from 'react-testing-library'

// Import de Theme Provider
import { ThemeProvider } from 'styled-components'
import Theme from 'common/Theme'

import DrawingsPreview from '../index'

describe('<DrawingsPreview />', () => {
  const renderComponent = (props = {}) => (
    render(
      <ThemeProvider theme={Theme}>
        <DrawingsPreview {...props} />
      </ThemeProvider>,
    )
  )

  beforeEach(() => {

  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    const id = 'id'
    const drawing = 'drawing'

    renderComponent({ id, drawing })

    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect log an error in console if called without id prop', () => {
    const spy = jest.spyOn(global.console, 'error')
    const drawing = 'drawing'

    renderComponent({ drawing })

    expect(spy).toHaveBeenCalledTimes(1)
  })

  it('Expect log an error in console if called without drawing prop', () => {
    const spy = jest.spyOn(global.console, 'error')
    const id = 'id'

    renderComponent({ id })

    expect(spy).toHaveBeenCalledTimes(1)
  })

  it('Should render and match the snapshot with default props', () => {
    const id = 'id'
    const drawing = 'drawing'

    const {
      container: { firstChild },
    } = renderComponent({ id, drawing })

    expect(firstChild).toMatchSnapshot()
  })

  it('Should render and match the snapshot with all props', () => {
    const id = 'id'
    const drawing = 'drawing'
    const selectMode = true
    const selected = true
    const draft = true
    const onClick = () => {}

    const {
      container: { firstChild },
    } = renderComponent({ id, drawing, selectMode, selected, draft, onClick })

    expect(firstChild).toMatchSnapshot()
  })

  it('Should call onClick props with given id prop as param', () => {
    const id = 'id'
    const drawing = 'drawing'
    const onClick = jest.fn()

    const {
      container: { firstChild },
    } = renderComponent({ id, drawing, onClick })
    fireEvent.click(firstChild)

    expect(onClick).toHaveBeenCalledTimes(1)
    expect(onClick).toHaveBeenCalledWith(id)
  })

})
