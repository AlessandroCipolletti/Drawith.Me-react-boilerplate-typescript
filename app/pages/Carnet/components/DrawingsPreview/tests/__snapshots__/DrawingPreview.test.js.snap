// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawingsPreview /> Should render and match the snapshot with all props 1`] = `
<div
  class="styled__Wrapper-piem58-0 kVBaoD"
  data-id="id"
>
  <div
    class="styled__DraftLabel-piem58-1 ddKMGt"
  />
  <div
    class="styled__SelectLabel-piem58-2 eonlas"
  />
</div>
`;

exports[`<DrawingsPreview /> Should render and match the snapshot with default props 1`] = `
<div
  class="styled__Wrapper-piem58-0 kVBaoD"
  data-id="id"
/>
`;
